### PR TITLE
Fixed login tests for both .net and node server

### PR DESCRIPTION
**Pull request by mamaso**

### CODE REVIEW

File: e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/tests/LoginTests.java [hunk 1]
```diff
@@ -75,12 +75,10 @@ public LoginTests(boolean isNetBackend) {
             this.addTest(createLoginTest(provider));
             this.addTest(createCRUDTest(USER_PERMISSION_TABLE_NAME, provider, TablePermission.User, true));
 
-            if (!isNetBackend) {
-                if (providersWithRecycledTokenSupport.contains(provider)) {
-                    this.addTest(createLogoutTest());
-                    this.addTest(createClientSideLoginTest(provider));
-                    this.addTest(createCRUDTest(USER_PERMISSION_TABLE_NAME, provider, TablePermission.User, true));
-                }
+            if (providersWithRecycledTokenSupport.contains(provider)) {
```
Comment (phvannor) [inline e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/tests/LoginTests.java]: Twitter and MSA can be used here too (see: https://github.com/Azure/azure-mobile-apps-ios-client/blob/master/e2etest/ZumoE2ETestApp/ZumoLoginTests.m#L54)  Also google works on Android platform only, so should also be able to be made to work here.

**Comment by phvannor** [comment]
Can you make the logout test more robust as well?  iOS had a cookie issue that would not be caught by logout test as is here.

File: e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/tests/LoginTests.java [hunk 2]
```diff
@@ -75,12 +75,10 @@ public LoginTests(boolean isNetBackend) {
...
+                this.addTest(createLogoutTest());
```
Comment (phvannor) [inline e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/tests/LoginTests.java]: Not important right now, but just calling out that long term this pattern is invalid, logout will actually invalidate the token used to login, so this test will need revisited.  We should probably set client.user = null instead but that can be done later on.

File: e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/tests/LoginTests.java (hunk 1 above)
Comment (mamaso) [inline e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/tests/LoginTests.java]: Ok good stuff, I'll see if I can extend client side login with those providers

**Comment by mamaso** [comment]
Is that cookie test here? https://github.com/Azure/azure-mobile-apps-ios-client/blob/master/e2etest/ZumoE2ETestApp/ZumoLoginTests.m#L269

File: e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/tests/LoginTests.java (hunk 2 above)
Comment (mamaso) [inline e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/tests/LoginTests.java]: good note, thanks

**Comment by phvannor** [comment]
That one is clearing provider cookies so that the UI is shown the next run.  The cookie test is these two sets: https://github.com/Azure/azure-mobile-apps-ios-client/blob/master/e2etest/ZumoE2ETestApp/ZumoLoginTests.m#L71-L72

**Comment by mamaso** [comment]
Added issues #13, #14, #15 for @phvannor 

I want to merge this PR and fix those down the track

**Pull request merged**